### DISCUSSION TITLE
Stitching images to create a map useing ImageMagick

### DISCUSSION THREAD

**Discussion by Stevieb36** [ImageMagick/ImageMagick#2446] [Help]
I am new to ImageMagick and using the Windows version via the command line. I would like to stitch an 8x8 matrix of images (all high resolution) together to create a large map in the following matrix:
A1.png  A2.png  A3.png  A4.png  A5.png  A6.png  A7.png  A8.png
B1.png  B2.png  B3.png  B4.png  B5.png  B6.png  B7.png  B8.png
C1.png  C2.png  C3.png  C4.png  C5.png  C6.png  C7.png  C8.png
D1.png  D2.png  D3.png  D4.png  D5.png  D6.png  D7.png  D8.png
E1.png  E2.png  E3.png  E4.png  E5.png  E6.png  E7.png  E8.png
F1.png  F2.png  F3.png  F4.png  F5.png  F6.png  F7.png  F8.png
G1.png  G2.png  G3.png  G4.png  G5.png  G6.png  G7.png  G8.png
H1.png  H2.png  H3.png  H4.png  H5.png  H6.png  H7.png  H8.png

I have used the following script to test how this would work, and I ended up with a low resolution composite with a border around each image.  
magick montage -adjoin A1.png a2.png test1.png

My ultimate goal is to keep the original resolution and have no border between the images. Any advice on how to learn more or what script I should use would be appreciative. I would be happy to modify the image file names if this makes it easier to write the script.

**Comment by snibgo** [comment]
Assuming you have no other files in that directory, and assuming all images are the same size, I would do it by horizontally appending the images that make each row, then appending those vertically. Like this:
```
magick ^
  ( A*.png +append +repage ) ^
  ( B*.png +append +repage ) ^
  ( C*.png +append +repage ) ^
  ( D*.png +append +repage ) ^
  ( E*.png +append +repage ) ^
  ( F*.png +append +repage ) ^
  ( G*.png +append +repage ) ^
  ( H*.png +append +repage ) ^
  -append ^
  out.png
```
Montage does have options that won't resize images and create borders, if you prefer.

**Comment by GeeMack** [comment]
If they are the only *.png images in the directory, and the images are the same dimensions, and named in a way that keeps them in the order you want, the images can be easily assembled into an 8x8 grid with a simple command like this...

     magick *.png +append -crop 8x1@ -append result.png

That appends all the images into a horizontal row, crops that row into 8 equal pieces, then appends those pieces vertically. If you're doing any additional processing in the command you'll probably want to "+repage" after that last "-append" to reset the image paging geometry.

**Comment by fmw42** [comment]
As GeeMack points out, if you want them tiled as alphabetically listed, then for IM 7, just use

`magick montage *.png -tile 8x8 -geometry +0+0 result.png`

If on IM 6, then remove magick from in front of montage

**Comment by Stevieb36** [comment]
Thank you all, it worked.
If you can point me in the direction of where (user guide, etc.) I can learn more about Image Magick that would be most appreciative.

**Reply by fmw42** [reply]
http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=9620
http://www.imagemagick.org/script/command-line-processing.php
https://imagemagick.org/script/command-line-options.php
http://www.imagemagick.org/Usage/reference.html
http://www.imagemagick.org/Usage/
https://github.com/ImageMagick/usage-markdown
https://imagemagick.org/script/porting.php#cli
